src/Recruit.Storage.Client/Core/Mongo/MongoDbCollectionBase.cs

```diff
@@ -0,0 +1,46 @@
+﻿using Microsoft.Extensions.Options;
+using MongoDB.Driver;
+using System;
+using System.Collections.Generic;
+using System.Security.Authentication;
+using System.Text;
+
+namespace Esfa.Recruit.Storage.Client.Core.Mongo
+{
+    public abstract class MongoDbCollectionBase
+    {
+        private readonly string _dbName;
+        private readonly string _collectionName;
+        private readonly MongoDbConnectionDetails _details;
+
+        private const string MongoCredentialMechanism = "SCRAM-SHA-1";
+
+        public MongoDbCollectionBase(string dbName, string collectionName, IOptions<MongoDbConnectionDetails> details)
+        {
+            _dbName = dbName;
+            _collectionName = collectionName;
+            _details = details.Value;
+        }
+
+        protected IMongoCollection<T> GetCollection<T>()
```

Comment:
I'm wondering if it would make things simpler if we create the client from a mongo connection string:

```
string connectionString = 
  @"mongodb://<user>:<password>@<host>:<port>/?ssl=true&replicaSet=globaldb";

MongoClientSettings settings = MongoClientSettings.FromUrl(
  new MongoUrl(connectionString)
);
```